external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/JsonUtils.scala

```diff
@@ -76,6 +76,16 @@ private object JsonUtils {
     }
   }
 
+  def topicTimestamps(str: String): Map[String, Long] = {
+    try {
+      Serialization.read[Map[String, Long]](str)
+    } catch {
+      case NonFatal(x) =>
+        throw new IllegalArgumentException(
+          s"""Expected e.g. {"topicA": 1549597128110,"topicB": 1549597120110}, got $str""")
```

Comment:
I considered actual use cases and feel we don't have to let end users describe all partitions. Most of use cases end users wouldn't want to set different timestamp for partitions. They mostly want to rewind to specific timestamp and replay, which timestamp should be global here.
Maybe we even don't want to set it for each topic too, but I just open the possibility. If we agree timestamp should be same for topics as well I can change it. I'm still open to support timestamp to partition level, but would like to hear more voices on this.

Reply:
> Most of use cases end users wouldn't want to set different timestamp for partitions.

I agree with this but the question is why such functionality provided in case of offsets? Per partition offset smells similar to per partition timestamp.

Reply:
Read offsets for each partition in specific time would be different for most of cases, so twos are different. My 2 cents, I think referring offsets was mostly needed when Kafka didn't have timestamp as index. Once we provide feature on timestamp index end users would love to use it instead of offset, except someone who know the exact offsets where to replaying.

Reply:
> I think referring offsets was mostly needed when Kafka didn't have timestamp as index

Agree but such example scenario would force user to fall back again: Given a cluster where there is no time sync because admins forgot it. Huge amount of messages generated with bad local time but consistently within partitions. User knows the time diffs but such case offsets has to be used. Let's hear others whether it worth.